impl_core/src/test/java/io/opencensus/implcore/metrics/LongGaugeImplTest.java

```diff
@@ -0,0 +1,166 @@
+
+/** Unit tests for {@link LongGaugeImpl}. */
+@RunWith(JUnit4.class)
+public class LongGaugeImplTest {
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  private static final String METRIC_NAME = "name";
+  private static final String METRIC_DESCRIPTION = "description";
+  private static final String METRIC_UNIT = "1";
+  private static final LabelKey LABEL_KEY = LabelKey.create("key", "key description");
+  private static final LabelValue LABEL_VALUES = LabelValue.create("value");
+  private static final LabelValue LABEL_VALUES_1 = LabelValue.create("value1");
+  private final List<LabelKey> labelKeys = new ArrayList<LabelKey>();
+  private final List<LabelValue> labelValues = new ArrayList<LabelValue>();
+  private final List<LabelValue> labelValues1 = new ArrayList<LabelValue>();
+
+  private static final Timestamp TEST_TIME = Timestamp.create(1234, 123);
+  private final TestClock testClock = TestClock.create(TEST_TIME);
+
+  private LongGaugeImpl longGaugeMetric;
+
+  @Before
+  public void setUp() {
+    labelKeys.add(LABEL_KEY);
+    labelValues.add(LABEL_VALUES);
+    labelValues1.add(LABEL_VALUES_1);
+
+    longGaugeMetric = new LongGaugeImpl(METRIC_NAME, METRIC_DESCRIPTION, METRIC_UNIT, labelKeys);
+  }
+
+  @Test
+  public void empty_GetMetrics() {
+    assertThat(longGaugeMetric.getMetric(testClock)).isEqualTo(null);
+  }
+
+  @Test
+  public void getOrCreateTimeSeries_WithLabels() {
+    LongPoint point = longGaugeMetric.getOrCreateTimeSeries(labelValues);
+    point.add(100);
+    point.add(-60);
+    point.set(500);
+
+    assertThat(longGaugeMetric.getMetric(testClock))
+        .isEqualTo(
+            Metric.create(
+                MetricDescriptor.create(
+                    METRIC_NAME,
+                    METRIC_DESCRIPTION,
+                    METRIC_UNIT,
+                    Type.GAUGE_INT64,
+                    Collections.singletonList(LABEL_KEY)),
+                Collections.singletonList(
+                    TimeSeries.createWithOnePoint(
+                        Collections.singletonList(LABEL_VALUES),
+                        io.opencensus.metrics.export.Point.create(Value.longValue(500), TEST_TIME),
+                        null))));
+  }
+
+  @Test
+  public void getDefaultTimeSeries() {
+    LongPoint point = longGaugeMetric.getDefaultTimeSeries();
+    point.add(100);
+    point.set(500);
+
+    LongPoint point1 = longGaugeMetric.getDefaultTimeSeries();
+    point1.add(-100);
+
+    assertThat(longGaugeMetric.getMetric(testClock))
+        .isEqualTo(
+            Metric.create(
+                MetricDescriptor.create(
+                    METRIC_NAME,
+                    METRIC_DESCRIPTION,
+                    METRIC_UNIT,
+                    Type.GAUGE_INT64,
+                    Collections.singletonList(LABEL_KEY)),
+                Collections.singletonList(
+                    TimeSeries.create(
+                        Collections.<LabelValue>emptyList(),
```

Comment:
This is incorrect. See comment in the ctor of the class why. default LabelValue does not mean missing value.